app/models/user.rb

```diff
@@ -1,7 +1,33 @@
 class User < ActiveRecord::Base
   has_many  :reviews
-  has_many  :queue_items
+  has_many  :queue_items, -> { order(order: :asc) }
+  
   validates_uniqueness_of :email
   validates_presence_of :email, :name
+  
   has_secure_password
+
+  def normalize_order
+    queue_items.each_with_index do |item, index|
+      item.update(order: index + 1)
+    end
+  end
+
+  def next_item_order
+    queue_items.count + 1
+  end
+
+  def update_queue(params)
```

Comment:
Pushing this down to the model layer is actually considered as bad practice. I am referring to [this assignment](https://www.gotealeaf.com/lessons/00bcc3b5/assignments/1569) especially solution video from 0:17 to 1:07.